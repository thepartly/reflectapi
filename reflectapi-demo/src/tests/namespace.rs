async fn empty<S>(_s: S, _: reflectapi::Empty, _: reflectapi::Empty) -> reflectapi::Empty {
    reflectapi::Empty {}
}

#[test]
fn test_namespace_with_dash() {
    assert_builder_snapshot!(reflectapi::Builder::<()>::new()
        .name("pet-orders".into())
        .route(empty, |b| {
            b.name("jobs-two.pet-orders.list".into())
                .description("desc".into())
        }))
}
