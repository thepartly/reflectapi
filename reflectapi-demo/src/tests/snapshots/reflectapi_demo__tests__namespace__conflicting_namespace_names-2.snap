---
source: reflectapi-demo/src/tests/namespace.rs
expression: typescript
snapshot_kind: text
---
// DO NOT MODIFY THIS FILE MANUALLY
// This file was generated by reflectapi-cli
//
// Schema name: foo
//

export function client(base: string | Client): __definition.Interface {
  return __implementation.__client(base);
}

export namespace __definition {
  export interface Interface {
    x: XInterface;
    y: YInterface;
  }

  export interface XInterface {
    foo: XFooInterface;
  }

  export interface XFooInterface {
    get: (input: {}, headers: {}) => AsyncResult<{}, {}>;
  }

  export interface YInterface {
    foo: YFooInterface;
  }

  export interface YFooInterface {
    get: (input: {}, headers: {}) => AsyncResult<null, {}>;
  }
}
export namespace reflectapi {
  /**
   * Struct object with no fields
   */
  export interface Empty {}

  /**
   * Error object which is expected to be never returned
   */
  export interface Infallible {}
}

namespace __implementation {

  function x__foo__get(client: Client) {
    return (input: {}, headers: {}) =>
      __request<{}, {}, {}, {}>(client, "/x.foo.get", input, headers);
  }
  function y__foo__get(client: Client) {
    return (input: {}, headers: {}) =>
      __request<{}, {}, null, {}>(client, "/y.foo.get", input, headers);
  }
}
